### PR TITLE
Support deploying smart contract tokens (ERC20)

### PR DESCRIPTION
**Pull request by alex-stone**
### What changed? Why?

This adds support for deploying fungible tokens (ERC20s) using MPC wallets

Example deployed ERC20 Token Contract: https://sepolia.basescan.org/address/0x20f74fbccfa248334972345b0c081a8fbff73f3b

#### Example
```ruby
# Create a base-sepolia Wallet
wallet = Coinbase::Wallet.create

# Fund the wallet (Or send funds to wallet.default_address on base-sepolia)
wallet.faucet

# Deploy an ERC20 Token
token = wallet.default_address.deploy_token(name: "CDP Test", symbol: "CDP", total_supply: 1_000_000_000)

token.wait!

# Transfer some of the the newly deployed ERC20 token.
wallet.transfer(1_000, token.contract_address, wallet.create_address)
```

#### Qualified Impact

This is a new feature

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: lib/coinbase/smart_contract.rb [hunk 1]
```diff
@@ -23,63 +23,64 @@ def self.list_events(
...
-    end
+    # Creates a new ERC20 token contract, that can subsequently be deployed to
+    # the blockchain.
+    # @param address_id [String] The address ID of deployer
+    # @param wallet_id [String] The wallet ID of the deployer
+    # @param name [String] The name of the token
+    # @param symbol [String] The symbol of the token
+    # @param total_supply [String] The total supply of the token, denominate in whole units.
+    # @return [SmartContract] The new ERC20 Token SmartContract object
+    def self.create_token_contract(
```
Comment (rohan-agarwal-coinbase) [inline lib/coinbase/smart_contract.rb:50]: in the Node SDK, i put this same code in `wallet_address.ts`. Should either this change or the Node SDK change or is ok that the code lives in different places?

**Review by rohan-agarwal-coinbase — APPROVED** [review]
PR should target `v0.6.0` instead of `master`. otherwise looks good!

File: lib/coinbase/smart_contract.rb (hunk 1 above)
Comment (alex-stone) [reply lib/coinbase/smart_contract.rb:50]: Good question, in ruby this always lives with the model.

In NodeJS, it has lived on the other model (which I think isn't ideal here, but is a bigger change).

**Pull request merged**